pyproject.toml

```
@@ -1,12 +1,67 @@
 [build-system]
-requires = [
-    "setuptools>=42",
-    "setuptools_scm[toml]>=3.4",
-    "scikit-build",
-    "cmake",
-    "ninja; platform_system!='Windows'"
+requires = ["scikit-build-core[pyproject]"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "pylibmagic"
+dynamic = ["version"]
+description = "scikit-build project with CMake for compiling libmagic"
+readme = "README.rst"
+requires-python = ">=3.7"
+authors = [
+    { name = "Giordon Stark", email = "kratsg@gmail.com" },
+]
+classifiers = [
+    "Development Status :: 1 - Planning",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Scientific/Engineering",
+]
+dependencies = [
+    "importlib-resources;python_version<'3.9'",
+    "typing-extensions>=3.7;python_version<'3.8'",
+]
+
+[project.optional-dependencies]
+dev = [
+    "pytest-cov>=3",
+    "pytest>=6",
+    "tbump>=6.7.0",
+    "twine",
 ]
-build-backend = "setuptools.build_meta"
+docs = [
+    "myst-parser>=0.13",
+    "sphinx-book-theme>=0.1.0",
+    "sphinx-copybutton",
+    "Sphinx>=4.0",
+]
+test = [
+    "pytest-cov>=3",
+    "pytest>=6",
+    "python-magic",
+]
+
+[project.urls]
+"Bug Tracker" = "https://github.com/kratsg/pylibmagic/issues"
+Changelog = "https://github.com/kratsg/pylibmagic/releases"
+Discussions = "https://github.com/kratsg/pylibmagic/discussions"
+Documentation = "https://pylibmagic.readthedocs.io/"
+Homepage = "https://github.com/kratsg/pylibmagic"
+
+[tool.scikit-build]
+metadata.version.provider = "scikit_build_core.metadata.setuptools_scm"
+sdist.include = ["src/pylibmagic/_version.py"]
+wheel.install-dir = "pylibmagic"
+minimum-version = "0.3"
```

```suggestion
minimum-version = "0.3"
wheel.py-api = "py3"
wheel.expand-macos-universal-tags = false
```
 cibuildwheel automatically avoids building the same wheel over again, so you can leave it checking the other Python versions if you want. Though you should not build x86 & arm macOS wheels, as the universal one will be expanded to include all tags.

---

I'm not sure I follow here, what do you mean?

---

I can try adjusting it tomorrow. Is there a way to run the CI to test wheel builds (I know how with GHA)? Is there a reason cirrus is used for almost everything?

FYI, this will speed up wheel building by 5x-8x, since you don’t need to build one for every Python version.

---

Cirrus is running:

```
pylibmagic-0.1.dev132+g4f874e2-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl   605 kB
  pylibmagic-0.1.dev132+g4f874e2-py3-none-musllinux_1_1_x86_64.whl
```

---

I can give you developer access so you can run the wheels manually. But I was using Cirrus primarily for the arm64 builds -- i just noticed it was significantly faster for the x86 builds compared to GHA too.